sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -551,6 +586,504 @@ Parameters:
+in all registered graph nodes.
+* `newAccessRange` - Access range for the accessor of the new buffer parameter.
+* `newAccessOffset` - Access offset for the range of the new buffer parameter.
+
+Exceptions:
+
+Throws an exception with the errc::invalid error code if the sum of
+newAccessRange and newAccessOffset exceeds the range of bufferRef in any dimension.
+
+
+|===
+
+===== The dynamic_local_accessor class [[dynamic-local-accessor-class]]
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+template <typename DataT, int Dimensions = 1>
+class dynamic_local_accessor {
+public:
+  dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        const property_list &propList = {});
+
+  dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        range<Dimensions> allocationSize,
+        const property_list &propList = {});
+
+  void update(range<Dimensions> newAllocationSize);
+};
+}
+----
+
+The `dynamic_local_accessor` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_local_accessor` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_local_accessor(
+      command_graph<graph_state::modifiable> graph,
+      const property_list &propList = {});
+----
+|Available only when (Dimensions == 0).
+
+Constructs a dynamic local accessor for accessing a single `DataT` element.
+This object can be registered with command graph nodes.
+
+Parameters:
+
+* `graph` - Graph which will contain the nodes that use the dynamic local accessor.
+
+|
+[source,c++]
+----
+template <typename DataT, int Dimensions = 1>
+dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        range<Dimensions> allocationSize);
+----
+|Available only when (Dimensions > 0).
+
+Constructs a dynamic local accessor object that can be registered with command graph.
+
+Parameters:
+
+* `graph` - Graph which will contain the nodes that use the dynamic local accessor.
+* `allocationSize` - The size of the local accessor.
+
+|
+[source,c++]
+----
+void update(range<Dimensions> newAllocationSize);
+----
+
+|Updates the allocation size of this dynamic local accessor in all graph nodes
+where it is registered. This new size will be reflected immediately in the
+modifiable graph which contains the registered nodes. The new size will not be
+reflected in any executable graphs created from that modifiable graph until
+`command_graph::update()` is called passing the modified nodes, or a new
+executable graph is finalized from the modifiable graph.
+
+It is not an error if `newAllocationSize` is set to the current allocation size
+of any registered nodes.
+
+Parameters:
+
+* `newAllocationSize` - The new allocation size that this dynamic local accessor will
+use in all the graph nodes where it is registered.
+
+|===
+
+===== The dynamic_work_group_memory class [[dynamic-work-group-memory-class]]
+
```

Comment:
Minor but a quick sentence linking to the `sycl_ext_oneapi_work_group_memory` extension doc again here would be handy.